### PR TITLE
Added in a check to ensure that a page has not already been called

### PR DESCRIPTION
**Pull request by gin0115**
Fixes an issue where infinite scroll would duplicate some posts due to the same AJAX call being executed twice. This was likely caused by a race condition where the IntersectionObserver triggered again before the previous request returned.

This PR includes:

- Changes to how infinite scroll is triggered: we now log the next page number before making the AJAX call and check if it has already been requested.

### CODE REVIEW

File: assets/js/src/frontend.js [hunk 1]
```diff
@@ -8,7 +12,14 @@ const intersectionObserver = new IntersectionObserver( ( entries ) => {
 			.querySelector( '.wp-block-post-template' ),
 		$clickedButton = entries[ 0 ].target;
 
-	fetchPosts( $url, $container, $clickedButton );
+	// Get the page id from url arg page=n
+	const page = $url.match( /page=(\d+)/ );
```
Comment (fmfernandes) [inline assets/js/src/frontend.js]: Just for the sake of not using Regex and maybe also improving readability what if we tried to get the page param with something like:

```suggestion
	const page = new URL( $url ).searchParams.get( 'page' )
```

**Review by fmfernandes — COMMENTED** [review]
PR looks good to me. I left just a minor suggestion.

File: assets/js/src/frontend.js (hunk 1 above)
Comment (gin0115) [reply assets/js/src/frontend.js]: This won't work as it's not a true url arg

`https://some.tld/post/?query-100-page=4`

However this get me to thinking i am not handling multiple queries per page, so i have made some changes to keep track of both the query id and the page number.

I have made changes to the PR and tested it the best (the plugin isnt the best with multiple queries per page)

**Comment by gin0115** [comment]
Many thanks :)

**Pull request merged**